GoogleForCreators/web-stories-wp
# Add srcset attribute for images

## Summary

This PR introduces the srcset attribute for all images (based on a resource) displayed in the editor.

## Relevant Technical Choices

The sizes attribute is omitted, because the size of image relative to the viewport cannot be determined. Given each of these images has an explicit size, the browser is able to obtain the right size image to display.

In the case of output.js, the amp-img tag uses the 'fill' layout, which with explicit width and height correctly autogenerates a valid 'sizes' attribute for the img tag.

## To-do

In the editor, an intermediate image size is used instead of the correct one according to the size of the image in the canvas. I believe this is because the editor adds the img tag and then resizes it at some point in the codebase.

This PR doesn't introduce the possibility of cropping the image when it's from Unsplash by using imgix params such as `rect`. I spent quite some time trying to get that to play nicely with how we display the image the AMP document and failed, but given that Unsplash are [not committed to supporting that param](https://unsplash.com/documentation#supported-parameters), I believe this improvement is enough for now.

## User-facing changes

No user visible changes should be seen, other than images (especially from Unsplash) displaying much faster and using much less bandwidth.

## Testing Instructions

Use the editor (especially by adding images from Unsplash via the `media3pTab` flag) and ensure that images display correctly in the editor and in the generated AMP stories.

---

Fixes #3249
